### ISSUE TITLE
Allow connector to overwrite "Streams" name

### ISSUE THREAD

**Issue by timroes** [airbytehq/airbyte#11100]
In https://github.com/airbytehq/airbyte/issues/11097 the suggestion was brought up that we don't want to use the name "Streams" in the UI for the Salesforce connector to do some more Salesforce specific name. Since this is currently hard coded in the UI, the ability we'd need for that is to have a connector overwrite that wording via it's specification.

This will though bring a couple of problems:

* The streams selection is linked to a connection, so which side of the connection (source or destination) should be able to overwrite that name? In the above example it would be the source, but Destinations might also use very specific naming for what a stream represents in it, and thus could also overwrite it? Though that would cause the problem, which wording to take if both source/destinations are overwriting it?
* We use "Streams" not just in the header of the box but within a lot of descriptions (within the UI and within the docs). Even if the connector specifies it custom wording for it, we can't fill it this word in all the descriptions, since all the texts are internationalizable and we can't properly fill in an English word we get from a connector into all of them. Thus I only see two options: We would only exchange the very header of the form to say that word and all other texts would still use the word "streams" in it (which I think is super confusing), or the connector would need to deliver us ALL those texts in the UI which contain "Streams", which imho is a really non-ideal API, and will make maintanance and changing any of those texts significantly harder in the future. It also doesn't solve the problem that we'll talk in our documentation still about "streams".
* Changing between different sources/destinations might confuse the user now more, since we have inconsistent wording between those.

Thus I don't think we should have the connectors being able to overwrite the word "Streams". Instead I would suggest that we add an in-product help modal via a help icon beside the "Streams" header, that will explain the user what "Streams" are in with a couple of examples, highlighting common use-cases like Objects in SalesForce or tables in a database.

@nataliekwong @sherifnada @misteryeo I would like to use this issue for discussing this, and try to get to a decicion how we'll handle this.

**Comment by octavia-squidington-iii** [comment]
cc @airbytehq/frontend

**Comment by nataliekwong** [comment]
I agree with being consistent throughout, if we give users the right guidance.  The solution to introduce an information icon is a fair solution to me.  

My suggestion for the help text: "A stream is an object in Salesforce"

**Comment by misteryeo** [comment]
Yep, I'd agree with consistency here - we shouldn't custom modify core parts of the UI for individual connectors. The proposed solution sounds right to me as well.

**Comment by michel-tricot** [comment]
@nataliekwong: "A stream is a type of object in Salesforce"

**Comment by timroes** [comment]
Blocked on https://github.com/airbytehq/airbyte/issues/14340
